This patch has already been committed to gnome-2-24 and trunk.

--- origsrc/evince-2.24.1/backend/dvi/mdvi-lib/t1.c	2008-10-20 14:50:53.000000000 -0500
+++ src/evince-2.24.1/backend/dvi/mdvi-lib/t1.c	2008-11-10 01:04:05.407003500 -0600
@@ -184,7 +184,7 @@
 		T1_DeleteAllSizes(i);
 	/* reset device resolutions */
 	if(T1_SetDeviceResolutions((float)xdpi, (float)ydpi) < 0)
-		warning(_("(t1) failed to reset device resolution\n"));
+		mdvi_warning(_("(t1) failed to reset device resolution\n"));
 	else
 		DEBUG((DBG_TYPE1, 
 			"(t1) reset successful, new resolution is (%d, %d)\n",
@@ -212,7 +212,7 @@
 			info->fontname, info->encoding->name));
 		T1_DeleteAllSizes(info->t1id);
 		if(T1_ReencodeFont(info->t1id, info->encoding->vector) < 0)
-			warning(_("%s: could not encode font\n"), info->fontname);
+			mdvi_warning(_("%s: could not encode font\n"), info->fontname);
 	}
 	if(info->mapinfo.slant) {
 		DEBUG((DBG_TYPE1, "(t1) %s: slanting by %.3f\n", 
